src/OmniSharp.DotNetTest/VSTestManager.cs

```diff
@@ -302,7 +292,7 @@ public override RunTestResponse RunTest(string[] methodNames, string testFramewo
 
                             testName = testName.Trim();
 
-                            if (testName.Equals(methodName, StringComparison.Ordinal))
+                            if (Array.Exists(methodNames, methodName => testName.Equals(methodName, StringComparison.Ordinal)))
```

Comment:
Nit: If you want to remove the O(n) operation here, you could add the method names to a `HashSet<string>(StringComparison.Ordinal)` and test that for containership here.